### PR TITLE
Add Path.isExecutable

### LINKED ISSUE #591: Add Path.isExecutable
*Platform* (all, jvm, js, android): jvm
*Extension* (none, kotlin 1.3): none

## Code related feature
```
expect(Path.of(foo)).isExecutable()
```
In #112, we decided to leave out `#isExecutable` because we had no use case for it. I am now writing a test where I want to check that a file is executable.

### PR DESCRIPTION
**Pull request by jGleitz**
closes #591 
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt [hunk 1]
```diff
@@ -672,6 +658,169 @@ abstract class PathAssertionsSpec(
...
+        }
+
+        context("POSIX: not executable", skip = ifPosixNotSupported) {
+            val expectedPermissionHint = String.format(HINT_ACTUAL_POSIX_PERMISSIONS.getDefault(), "u=rw g=r o=")
+
+            it("throws an AssertionError for a file") withAndWithoutSymlink { maybeLink ->
+                val expectedTypeHint = String.format(
+                    FAILURE_DUE_TO_PERMISSION_FILE_TYPE_HINT.getDefault(),
+                    A_FILE.getDefault(),
+                    EXECUTABLE.getDefault()
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt:720]: maybe worth introducing two functions, this is repeated several times
- fileTypeHintFor(translatable: Translatable)
- directoryTypeHintFor(translatable: Translatable)

**Review by robstoll — DISMISSED** [review]

**Comment by robstoll** [comment]
@jGleitz LGTM only one suggestion. Up to you if you want to improve it in this or in a next PR.

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt (hunk 1 above)
Comment (jGleitz) [reply misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: Done

**Comment by jGleitz** [comment]
@robstoll I addressed it in this PR.

**Review by robstoll — APPROVED** [review]

**Pull request merged**